### PR TITLE
Add Python 3.11 and drop Django 3.0, 3.1

### PR DESCRIPTION
**Pull request by hugorodgerbrown**
Update to classifiers and CI to support 3.11.

At the same time I have deprecated Django 3.0, 3.1 and removed
the `compat` module as 3.2+ supports `JSONField`.

### CODE REVIEW

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -37,29 +37,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python: [3.8,"3.10"]
-        django: [32,40,41,main]
+        python: [3.8,"3.10", "3.11"]
```
Comment (hugorodgerbrown) [inline .github/workflows/tox.yml:40]: Add the new version to the CI matrix.

File: .github/workflows/tox.yml [hunk 2]
```diff
@@ -37,29 +37,12 @@ jobs:
...
-          # Set health checks to wait until postgres has started
-        options: >-
-          --health-cmd pg_isready
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
-        ports:
-          # Maps tcp port 5432 on service container to the host
-          - 5432:5432
-
```
Comment (hugorodgerbrown) [inline .github/workflows/tox.yml:62]: Postgres is no longer required in CI as we are dropping Django 3.0, 3.1 which only support a Postgres JSON field type. 3.2+ support `JSONField` across all backends, including SQLite, so we run against that.

File: pyproject.toml [hunk 3]
```diff
@@ -19,17 +17,18 @@ classifiers = [
...
 packages = [{ include = "appmail" }]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-django = "^3.0 || ^4.0"
+django = "^3.2 || ^4.0"
 
 [tool.poetry.dev-dependencies]
-black = {version = "*", allow-prereleases = true}
+black = "*"
```
Comment (hugorodgerbrown) [inline pyproject.toml:29]: This is old - Black now has a released version so don't need to specify pre-release.

File: pyproject.toml [hunk 4]
```diff
@@ -19,17 +17,18 @@ classifiers = [
...
 coverage = "*"
-flake8 = "^3.9"
+flake8 = "*"
```
Comment (hugorodgerbrown) [inline pyproject.toml:31]: Flake introduced a change that broke Bandit for a while, but this is now fixed.

**Comment by hugorodgerbrown** [comment]
@emab take a look a why the tests failed - we may have to wait a while, but there's no reason not to crack on and do the updates in advance.

**Comment by emab** [comment]
> @emab take a look a why the tests failed - we may have to wait a while, but there's no reason not to crack on and do the updates in advance.

Is this just that `tox` isn't set up to deal with the new verisons yet?

**Comment by hugorodgerbrown** [comment]
> Is this just that `tox` isn't set up to deal with the new versions yet?

No - it's GH - it doesn't have Python 3.11 available yet.

**Review by djm — APPROVED** [review]

**Pull request merged**